Installer should only write to the registry during the install phase

sredna opened the pull request:
.onInit should not write to the registry. If the user cancels the installer, no traces should be left behind.

donho commented:
@sredna 
Logically, your code makes sense and looks better. However, it doesn't work after my testing of this PR:
It seems `WriteRegStr HKLM "SOFTWARE\${APPNAME}" 'InstallerLanguage' '$Language'` doesn't take effect - after changing the language and finishing the installation, the next launch of the same installer takes always the same language before changing.

I suggest you always test your PR locally before submitting.

sredna commented:
@donho I did test it. I tested it in a 32-bit VM which might be why it worked for me.

The installer is a bit schizophrenic about where it writes `HKLM "Software\${APPNAME}"` in the registry, it wants the path in the 64-bit view and the language in the 32-bit view.

I fixed the PR and added a note as to why it needs to be in the 32-bit view.
